hibernate-rx-core/src/main/java/org/hibernate/rx/event/impl/DefaultRxLoadEventListener.java

```diff
@@ -74,27 +76,31 @@ public void onLoad(
 			throw new HibernateException( "Unable to locate persister: " + event.getEntityClassName() );
 		}
 
-		Object checkId = checkId( event, loadType, persister );
-		if ( checkId instanceof CompletionStage) {
+		CompletionStage<Void> checkId = checkId( event, loadType, persister );
+		if ( !checkId.toCompletableFuture().isDone() ) {
```

Comment:
@FroMage Mentioned in the past that there might be some problems with this operation (`toCompletableFuture()`) and some implementations could throw an exception. @FroMage Do I remember correctly?

Reply:
I think it's OK, since we're only going to call it on a `CompletionStage` that we're created ourselves using `RxUtil.completedFuture()` or `RxUtil.nullFuture()`. We're never going to call it on a "real" `CompletionStage` that's coming from the JDBC driver.

Reply:
I don't remember mentioning that warning. I do that sort of stuff in several places and don't know of any better way to see if a `CS` is done. IMO that's OK.